src/main/java/org/cru/godtools/domain/packages/PageStructure.java

```diff
@@ -143,10 +164,161 @@ public void mergeXmlContent(Document updatedPageLayout)
 				}
 			}
 		}
-
 		xmlContent = updatedPageLayout;
 	}
 
+	public void addXmlContent(Document addXmlContentDocument) throws IOException, TransformerException
+	{
+		XmlUtilities.verifyDifferentXml(xmlContent, addXmlContentDocument);
+
+		NodeList addXmlNodeList = addXmlContentDocument.getElementsByTagName(ALL_ELEMENTS);
+		NodeList currentXmlContentNodeList = xmlContent.getElementsByTagName(ALL_ELEMENTS);
+
+		List<String> currentXmlStringListOfElements = Lists.newArrayList();
+
+		//create a string list of xml elements of the current document
+		for(int i=0; i < currentXmlContentNodeList.getLength(); i++)
+		{
+			String nodeString = XmlUtilities.xmlDocumentOrNodeToString(new DOMSource(currentXmlContentNodeList.item(i)));
+			currentXmlStringListOfElements.add(nodeString);
+		}
+
+		for(int i = 0; i < addXmlNodeList.getLength(); i++)
+		{
+			Node nodeToAdd = addXmlNodeList.item(i);
+			Node xmlContentNode = currentXmlContentNodeList.item(i);
+
+			if (xmlContentNode != null)
+			{
+				Element originalElement = (Element) xmlContentNode;
+				Element addElement = (Element) nodeToAdd;
+
+				boolean attrMatch = hasSameAttributes(originalElement, addElement);
+
+				//if the xml attributes are not the same but the node
+				//name is the same it's probably a new node
+				if (!attrMatch && originalElement.getNodeName().equals(addElement.getNodeName()))
+				{
+					String nodeToBeAddedString = XmlUtilities.xmlDocumentOrNodeToString(new DOMSource(nodeToAdd));
+
+					if (!currentXmlStringListOfElements.contains(nodeToBeAddedString))
+					{
+						Node targetNode = nodeToAdd.cloneNode(true);
+						Node nodeToBeImported = xmlContent.importNode(targetNode, true);
+						xmlContent.getDocumentElement().insertBefore(nodeToBeImported, xmlContentNode.getPreviousSibling());
+					}
+				}
+			}
+			else
+			{
+				Node targetNode = xmlContent.importNode(nodeToAdd, true);
+				currentXmlContentNodeList.item(1).appendChild(targetNode);
+			}
+		}
+	}
+
+	private boolean hasSameAttributes(Element oElement, Element aElement)
+	{
+		NamedNodeMap originalNamedNodeMap = oElement.getAttributes();
+		NamedNodeMap additionNamedNodeMap = aElement.getAttributes();
+
+		boolean attrMatch = true;
+
+		for (int n = 0; n < additionNamedNodeMap.getLength(); n++)
+        {
+            Attr a1 = (Attr) additionNamedNodeMap.item(n);
+            Attr o1 = (Attr) originalNamedNodeMap.item(n);
+
+            if (!o1.getName().equals(a1.getName()) || !o1.getValue().equals(a1.getValue()))
+            {
+                attrMatch = false;
```

Comment:
Just `return false;` here.  Then you can do away with the `boolean attrMatch` declaration, and `return true;` at the end of the method.  Simplifies things a bit.